python/dgl/nn/pytorch/conv/gatconv.py

```diff
@@ -304,7 +304,8 @@ def forward(self, graph, feat, get_attention=False):
             rst = graph.dstdata['ft']
             # residual
             if self.res_fc is not None:
-                resval = self.res_fc(h_dst).view(h_dst.shape[0], self._num_heads, self._out_feats)
+                # Use -1 rather than self._num_heads to handle broadcasting
+                resval = self.res_fc(h_dst).view(h_dst.shape[0], -1, self._out_feats)
```

Comment:
Could you please explain why does it work?

Reply:
If we use `self._num_heads`, then we will get an error with `view`. If we use -1, it will be 1 in this case. We can add a tensor of shape `(A, 1, C)` to a tensor of shape `(A, B, C)`.